src/test/java/io/jenkins/plugins/grading/AnalysisScoreTest.java

```diff
@@ -45,4 +65,81 @@ void shouldConvertFromJson() {
         assertThat(configuration).hasLowImpact(4);
         assertThat(configuration).hasMaxScore(5);
     }
+
+    @Test
+    void shouldReturnPositiveParams() {
+
+        AnalysisResult result = mock(AnalysisResult.class);
+        when(result.getTotalErrorsSize()).thenReturn(3);
+        when(result.getTotalHighPrioritySize()).thenReturn(5);
+        when(result.getTotalNormalPrioritySize()).thenReturn(2);
+        when(result.getTotalLowPrioritySize()).thenReturn(4);
+        when(result.getId()).thenReturn("result-id");
+
+        AnalysisConfiguration configuration = mock(AnalysisConfiguration.class);
+        when(configuration.getErrorImpact()).thenReturn(1);
+        when(configuration.getHighImpact()).thenReturn(1);
+        when(configuration.getLowImpact()).thenReturn(1);
+        when(configuration.getNormalImpact()).thenReturn(1);
+
+        AnalysisScore analysisScore = new AnalysisScore("Results", configuration, result);
+
+        assertThat(analysisScore.getErrorsSize()).isEqualTo(3);
+        assertThat(analysisScore).hasErrorsSize(3);
+        assertThat(analysisScore).hasHighPrioritySize(5);
+        assertThat(analysisScore).hasNormalPrioritySize(2);
+        assertThat(analysisScore).hasLowPrioritySize(4);
+        assertThat(analysisScore).hasName("Results");
+        assertThat(analysisScore).hasId("result-id");
+    }
+
+    @Test
+    void shouldReturnNegativeParams() {
+
+        AnalysisResult result = mock(AnalysisResult.class);
+        when(result.getTotalErrorsSize()).thenReturn(-3);
+        when(result.getTotalHighPrioritySize()).thenReturn(-5);
+        when(result.getTotalNormalPrioritySize()).thenReturn(-2);
+        when(result.getTotalLowPrioritySize()).thenReturn(-4);
+        when(result.getId()).thenReturn("result-id");
+
+        AnalysisConfiguration configuration = mock(AnalysisConfiguration.class);
+        when(configuration.getErrorImpact()).thenReturn(1);
+        when(configuration.getHighImpact()).thenReturn(1);
+        when(configuration.getLowImpact()).thenReturn(1);
+        when(configuration.getNormalImpact()).thenReturn(1);
+
+        AnalysisScore analysisScore = new AnalysisScore("Results", configuration, result);
+
+        assertThat(analysisScore.getErrorsSize()).isEqualTo(-3);
+        assertThat(analysisScore).hasErrorsSize(-3);
+        assertThat(analysisScore).hasHighPrioritySize(-5);
+        assertThat(analysisScore).hasNormalPrioritySize(-2);
+        assertThat(analysisScore).hasLowPrioritySize(-4);
+        assertThat(analysisScore).hasName("Results");
+        assertThat(analysisScore).hasId("result-id");
+    }
+
+    @Test
+    void shouldReturnNullParams() {
```

Comment:
Da haben Sie aus meiner Sicht einen Bug entdeckt: in meinem Code ist `null` nicht erlaubt. Normalerweise fliegt eh eine NPE. Allerdings hier nicht: daher wäre es gut den Code anzupassen mit einem
```
Ensure.that(name).isNotEmpty();
Ensure.that(id).isNotEmpty();
```
im Constructor. Und dann im Test die IAE überprüfen...

Reply:
Ich habe gerade noch einen PR gemacht, der `null` besser berücksichtigt: https://github.com/jenkinsci/autograding-plugin/pull/10

Reply:
Ist eingefügt. Passt Ihnen der Testfall?